engine/Core.php

```diff
@@ -105,12 +117,21 @@ private function serve_child_module_asset($asset_path, $file_name) {
                     $content_type = mime_content_type($asset_path);
 
                     if ($content_type == 'text/plain') {
-
                         $pos2 = strpos($file_name, '.css');
                         if (is_numeric($pos2)) {
                             $content_type = 'text/css';
                         }
-                        
+                         $pos2 = strpos($file_name, '.js');
+                        if (is_numeric($pos2)) {
+                            $content_type = 'text/javascript';
+                        }
+                    }
+                    
+                    if ($content_type == 'text/html') {
```

Comment:
Added to return 'text/javascript' instead of 'text/html' when using string literals.